logicrunner/builtin/contract/member/member.go

```diff
@@ -259,40 +256,32 @@ func (m *Member) transferCall(params map[string]interface{}) (interface{}, error
 
 	return w.Transfer(amount, recipientReference)
 }
-func (m *Member) migrationCall(params map[string]interface{}) (interface{}, error) {
+func (m *Member) migrationCall(params map[string]interface{}) error {
 
 	amountStr, ok := params["amount"].(string)
 	if !ok {
-		return nil, fmt.Errorf("incorect input: failed to get 'amount' param")
+		return fmt.Errorf("incorect input: failed to get 'amount' param")
 	}
 
 	amount := new(big.Int)
 	amount, ok = amount.SetString(amountStr, 10)
 	if !ok {
-		return nil, fmt.Errorf("failed to parse amount")
+		return fmt.Errorf("failed to parse amount")
 	}
-
-	currentDateStr, ok := params["currentDate"].(string)
-	if !ok {
-		return nil, fmt.Errorf("incorect input: failed to get 'currentDate' param")
+	if amount.Cmp(big.NewInt(0)) != 1 {
+		return fmt.Errorf("amount must be greater than zero")
 	}
-
-	currentDate, err := helper.ParseTimestamp(currentDateStr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse 'currentDate': %s", err.Error())
-	}
-
-	txId, ok := params["txId"].(string)
+	txId, ok := params["ethTxHash"].(string)
 	if !ok {
-		return nil, fmt.Errorf("incorect input: failed to get 'txId' param")
+		return fmt.Errorf("incorect input: failed to get 'ethTxHash' param")
 	}
 
-	burnAddress, ok := params["burnAddress"].(string)
+	burnAddress, ok := params["migrationAddress"].(string)
 	if !ok {
-		return nil, fmt.Errorf("incorect input: failed to get 'burnAddress' param")
+		return fmt.Errorf("incorect input: failed to get 'migrationAddress' param")
 	}
 
-	return m.migration(txId, burnAddress, *amount, currentDate)
+	return m.depositMigration(txId, burnAddress, *amount)
```

Comment:
you should send just ref on big.Int. for memory economy
rename migrationCall to depositMigrationCall too

Reply:
ok.Thank you